spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -360,8 +360,10 @@ protected void doInit() {
 			ConverterRegistry converterRegistry =
 					(ConverterRegistry) conversionService;
 
-			converterRegistry.addConverter(new ClassToStringConverter());
-			converterRegistry.addConverter(new ObjectToStringConverter());
+			if (!conversionService.canConvert(Class.class, String.class)) {
```

Comment:
I have some concerns about this - particularly for back-porting.

See `ConvertersForPair.add()` - it adds these converters to the head of the list, thus overriding any existing converter that `canConvert()`.

I am not saying this behavior is correct, but I don't think we can backport this part. Even in 4.2, we might need to add a boolean to allow user converters to take priority over this inbuilts.